spiffe-helper_test.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"crypto/x509"
+	"github.com/spiffe/spire/proto/api/workload"
+	"github.com/spiffe/spire/test/util"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+)
+
+//Creates a Sidecar with a Mocked WorkloadAPIClient and tests that
+//running the Sidecar Daemon, when a SVID Response is sent to the
+//UpdateChan on the WorkloadAPI client, the PEM files are stored on disk
+func TestSidecar_RunDaemon(t *testing.T) {
+
+	tmpdir, err := ioutil.TempDir("", "test-certs")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	config := &SidecarConfig{
+		Cmd:                "echo",
+		CertDir:            tmpdir,
+		SvidFileName:       "svid.pem",
+		SvidKeyFileName:    "svid_key.pem",
+		SvidBundleFileName: "svid_bundle.pem",
+	}
+
+	updateMockChan := make(chan *workload.X509SVIDResponse, 1)
+	workloadClient := MockWorkloadClient{
+		mockChan: updateMockChan,
+	}
+
+	sidecar := sidecar{
+		config:            config,
+		workloadAPIClient: workloadClient,
+	}
+
+	go func() {
+		err = sidecar.RunDaemon()
```

Comment:
RunDeamon will never return and this goroutine is essentially leaked. See comment elsewhere about adding a context parameter.